tensorbay/utility/name.py

```diff
@@ -192,6 +192,12 @@ def __contains__(self, key: Any) -> bool:
         except (KeyError, TypeError):
             return False
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, self.__class__):
+            return False
+
+        return self._data.__eq__(other._data) and self._names.__eq__(other._names)
```

Comment:
just check `self._data` is enough.
`self._names` is just a cache for all names in `self._data`, `self._names` should always equal to `[item.name for item in self._name]`

Reply:
Would someone change `_data` directly instead of using `add`?

Reply:
visit private is not protected by our code.
if someone change `_data` directory, not only `__eq__` method will be broken but also other methods like auto-sort will also be broken.
So that there are not need to add extra logic for someone who breaks the basic rule.